v2/reader.go

```diff
@@ -68,17 +68,13 @@ func (r *Reader) readHeader() (err error) {
 	return
 }
 
-// CarV1ReaderAt provides an io.ReaderAt containing the CAR v1 dump encapsulated in this CAR v2.
-func (r *Reader) CarV1ReaderAt() io.ReaderAt {
-	return r.carv1SectionReader()
-}
-
-func (r *Reader) carv1SectionReader() *io.SectionReader {
+// CarV1Reader provides a reader containing the CAR v1 section encapsulated in this CAR v2.
+func (r *Reader) CarV1Reader() *io.SectionReader {
 	return io.NewSectionReader(r.r, int64(r.Header.CarV1Offset), int64(r.Header.CarV1Size))
 }
 
-// IndexReaderAt provides an io.ReaderAt containing the carbs.Index of this CAR v2.
-func (r *Reader) IndexReaderAt() io.ReaderAt {
+// IndexReader provides an io.Reader containing the carbs.Index of this CAR v2.
+func (r *Reader) IndexReader() io.Reader {
```

Comment:
Because `index.ReadFrom` takes an `io.Reader`. It could certainly take an `io.ReaderAt` I thought that might not be as easy to instantiate as `io.ReaderAt` for reading indices. The actual reader type under the hood here is `internalio.OffsetReader` which implements both but it is an internal type. What do you suggest?

Reply:
I think this depends on whether we want to read the whole thing, or opportunistically only read what we need. If we are reading the whole thing, then I agree Reader is fine. See the latest Slack thread.

Reply:
That makes sense thanks @mvdan